internal/peer/node/start.go

```diff
@@ -496,13 +496,9 @@ func serve(args []string) error {
 		HandlerRegistry: chaincodeHandlerRegistry,
 	}
 	var client *docker.Client
-	if coreConfig.VMDockerTLSEnabled {
-		client, err = docker.NewTLSClient(coreConfig.VMEndpoint, coreConfig.DockerCert, coreConfig.DockerKey, coreConfig.DockerCA)
-	} else {
-		client, err = docker.NewClient(coreConfig.VMEndpoint)
-	}
-	if err != nil {
-		logger.Panicf("cannot create docker client: %s", err)
+
+	if coreConfig.VMEndpoint != "" {
```

Comment:
shouldn't we also add a check for coreConfig.ExternalBuilders[] being empty ... we need either a VMEndpoint or a list of ExternalBuilders

Reply:
+1 at the very least we should print a warning indicating that the peer will not be able to execute chaincode.

I suppose it's possible that someone simply wants to stand up a peer to pull and validate blocks, though without a chaincode to access state, I don't see any real usefulness for this configuration.

Reply:
Done

Reply:
@jyellick one use case could be to construct state db for direct query against it?

Reply:
It's easy enough to create a no-op external builder -- for instance:

```
mkdir -p mybuilder/bin
ln -s /bin/false mybuilder/bin/detect
```

Now you have a functioning external builder that will never build anything, and you can successfully make your peer start.  If we find lots of users needing to do a workaround like this, we can look into relaxing the check, but we can always relax checks, but adding constraints later is hard.